examples/kind/konnectivity-agent-ds.yaml

```diff
@@ -0,0 +1,70 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: konnectivity-agent
+  namespace: kube-system
+  labels:
+    kubernetes.io/cluster-service: "true"
+---
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels:
+    addonmanager.kubernetes.io/mode: Reconcile
+    k8s-app: konnectivity-agent
+  namespace: kube-system
+  name: konnectivity-agent
+spec:
+  selector:
+    matchLabels:
+      k8s-app: konnectivity-agent
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        k8s-app: konnectivity-agent
+    spec:
+      priorityClassName: system-cluster-critical
+      tolerations:
+        - key: "CriticalAddonsOnly"
+          operator: "Exists"
+        - operator: "Exists"
+          effect: "NoExecute"
+      nodeSelector:
+        kubernetes.io/os: linux
+      containers:
+      - name: konnectivity-agent-container
+        image: gcr.io/k8s-staging-kas-network-proxy/proxy-agent:master
+        resources:
+          requests:
+            cpu: 50m
+          limits:
+            memory: 30Mi
+        command: [ "/proxy-agent"]
+        args: [
+          "--logtostderr=true",
+          "--ca-cert=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+          "--proxy-server-host=kind-control-plane",
+          "--proxy-server-port=8091",
+          "--service-account-token-path=/var/run/secrets/tokens/konnectivity-agent-token",
+          ]
+        livenessProbe:
+          httpGet:
+            scheme: HTTP
+            port: 8093
+            path: /healthz
+          initialDelaySeconds: 15
+          timeoutSeconds: 15
+        volumeMounts:
+          - mountPath: /var/run/secrets/tokens
+            name: konnectivity-agent-token
+      serviceAccountName: konnectivity-agent
+      volumes:
+      - name: konnectivity-agent-token
+        projected:
+          sources:
+          - serviceAccountToken:
+              path: konnectivity-agent-token
+              audience: system:konnectivity-server
```

Comment:
@jkh52 @liggitt what is the easiest way to set up is for testing despite how insecure it is?

Reply:
The examples in this directory are relatively stale, I would start from the manifests in k/k which are kept up to date and have active prow test coverage (see https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/manifests/konnectivity-server.yaml and https://github.com/kubernetes/kubernetes/blob/master/cluster/gce/addons/konnectivity-agent/konnectivity-agent-ds.yaml).

Reply:
what is the token used for?

Reply:
It looks like the token used for agent authentication.

But note that konnectivity-server can be configured to [skip agent authentication](https://github.com/kubernetes-sigs/apiserver-network-proxy/blob/master/cmd/server/app/server.go#L124)

Reply:
ok, I managed to get it working, I'm adding CI in github too, I want to validate that it workswith ipv6 only